relay-general/src/store/normalize.rs

```diff
@@ -118,28 +118,41 @@ impl<'a> NormalizeProcessor<'a> {
     }
 
     /// Validates the timestamp range and sets a default value.
-    fn normalize_timestamps(&self, event: &mut Event) -> ProcessingResult {
+    fn normalize_timestamps(
+        &self,
+        event: &mut Event,
+        meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
         let received_at = self.config.received_at.unwrap_or_else(Utc::now);
         event.received = Annotated::new(received_at);
 
+        let mut sent_at = None;
+
         event.timestamp.apply(|timestamp, meta| {
             if let Some(secs) = self.config.max_secs_in_future {
                 if *timestamp > received_at + Duration::seconds(secs) {
                     meta.add_error(ErrorKind::FutureTimestamp);
-                    return Err(ProcessingAction::DeleteValueSoft);
+                    sent_at = Some(*timestamp);
+                    return Ok(());
                 }
             }
 
             if let Some(secs) = self.config.max_secs_in_past {
                 if *timestamp < received_at - Duration::seconds(secs) {
                     meta.add_error(ErrorKind::PastTimestamp);
```

Comment:
I would like to get rid of this EventError variant.

Reply:
I'm afraid you can't remove these. They are also used in minidump processing and have special handling in the UI.

What if you don't serialize an error in `ClockDriftProcessor` for this case, instead? Basically instead of giving it a reason, tell it to skip the meta error.

Reply:
fixed